src/main/java/net/minecraftforge/event/world/DimensionPreloadEvent.java

```diff
@@ -0,0 +1,33 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+/**
+ * This event is fired when DimensionManager is starting to init a new dimension
+ * with {@link DimensionManager#initDimension(int)},
+ * <br>
+ * This event is {@link Cancelable}.<br>
+ * <br>
+ * event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ * special care should be used when canceling this event
+ * use it only when you know what you are doing
```

Comment:
I understand that, but the API is terrible.
If you want to allow replacing the world instance, add a setWorld to the event instead of making it cancelable.